### ISSUE TITLE
Network facts return documentation

### ISSUE THREAD

**Issue by jmcgill298** [ansible/ansible#36867]
##### ISSUE TYPE
 - Documentation Report

##### ANSIBLE VERSION

```
N/A
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

Documentation does not explain that return values are nested under `ansible_facts`

##### STEPS TO REPRODUCE

N/A

```yaml
N/A
```

##### EXPECTED RESULTS

I expected to have to register the results and then access the key/value pairs directly under my register value.
Expected:
```yaml
- ios_facts:
    register: "facts_data"

- debug:
    var: ios_facts.ansible_net_config
```
The above debug var actually requires `ios_facts.ansible_facts.ansible_net_config`, but is also not needed as the facts module adds the key/value pairs to hostvars. The Ansible way would be to just use `ansible_net_config`.

Suggest adding documentation that explains the return values are part of `ansible_facts` and how `ansible_facts` behaves:
```
"Common return values are documented here Return Values, the following are the fields are returned as part of ansible_facts, which do not require registering returns and automatically added to hostvars:
```

Suggest updating examples to include an example of using the return data:
```
- name: GATHER CONFIG DATA
  ios_facts:
    gather_subset:
      - "config"

- name: ENSURE OUTPUT DIRECTORY EXISTS
  file:
    state: "direcotry"
    path: "./project/output/configs"

- name: WRITE BACKUP TO FILE
  copy:
        content: "{{ ansible_net_config }}"
        dest: ""./project/output/configs/{{ inventory_hostname }}.cfg"
```
##### ACTUAL RESULTS

```

```

**Comment by gundalow** [comment]
Would it be OK to add a few lines in the `notes:` section of the `_facts` modules to make this clearer?

**Comment by jmcgill298** [comment]
IMO the way the return header reads now indicates that the values are returned by the module, when actually only `ansible_facts` is returned. But notes is still better than what is there now. What do you think about just saying the following values are returned to ansible_facts?

**Comment by dagwieers** [comment]
I think we need to make a clear distinction between **returned values** and **returned facts**.
And have separate sections for these. Both sections would make clear how to use the values/facts.
I would even consider adding new terminology to make this clearer.

BTW IMO the returned values sources should have been part of the documentation blob, just like the returned facts. Rather than its own section, but that's a whole other story :-)

**Comment by dagwieers** [comment]
BTW I plan on redesigning the Returned Values section of the documentation as I have been doing for the rest of the docs. Mostly to avoid this chaos: http://docs.ansible.com/ansible/devel/modules/win_disk_facts_module.html

And if you have some time, let me know what you think about this: https://github.com/ansible/ansible/pull/36901

**Comment by tyll** [comment]
About the expected result, shouldn't it be ```facts_data.ansible_net_config``` instead of ```ios_facts.ansible_net_config``` where you should expect the returned values? I agree with @dagwieers that using the term returned facts for the returned facts sounds good to distinguish them from the return values and mentioning that the facts are also added to the hostvars.

**Comment by dagwieers** [comment]
So I have this implemented now in #36943 . I'd like to know from participants in this thread what text do we want to add at the top of the **Returned Facts** section ?

The current one is probably not sufficiently concrete.

> Facts returned by this module are added/updated in the `hostvars` host facts and can be referenced by name just like any other host fact. They do not need to be registered in order to use them.

Do we want to add a link to documentation ?

![screenshot from 2018-03-06 00-32-26](https://user-images.githubusercontent.com/388198/37005851-efe98bec-20d5-11e8-86b8-ddb51ee2e4c8.png)
